integration/pvtdata/data_purge_test.go

```diff
@@ -0,0 +1,155 @@
+		testDir, err = ioutil.TempDir("", "purgedata")
+		Expect(err).NotTo(HaveOccurred())
+
+		config := nwo.ThreeOrgRaft()
+		network = nwo.New(config, testDir, nil, StartPort(), components)
+
+		network.GenerateConfigTree()
+		network.Bootstrap()
+
+		networkRunner := network.NetworkGroupRunner()
+		process = ifrit.Invoke(networkRunner)
+		Eventually(process.Ready(), network.EventuallyTimeout).Should(BeClosed())
+
+		orderer = network.Orderer("orderer")
+		network.CreateAndJoinChannel(orderer, "testchannel")
+		network.UpdateChannelAnchors(orderer, "testchannel")
+		network.VerifyMembership(
+			network.PeersWithChannel("testchannel"),
+			"testchannel",
+		)
+		nwo.EnableCapabilities(
+			network,
+			"testchannel",
+			"Application", "V2_0",
+			orderer,
+			network.PeersWithChannel("testchannel")...,
+		)
+
+		chaincode = &nwo.Chaincode{
+			Name:              "marblesp",
+			Version:           "0.0",
+			Path:              components.Build("github.com/hyperledger/fabric/integration/chaincode/marbles_private/cmd"),
+			Lang:              "binary",
+			PackageFile:       filepath.Join(testDir, "purgecc.tar.gz"),
+			Ctor:              `{"Args":[]}`,
+			SignaturePolicy:   `OR ('Org1MSP.member','Org2MSP.member', 'Org3MSP.member')`,
+			CollectionsConfig: CollectionConfig("collections_config1.json"),
+			Sequence:          "1",
+			InitRequired:      false,
+			Label:             "purgecc_label",
+		}
+
+		nwo.DeployChaincode(network, "testchannel", orderer, *chaincode)
+
+		org2Peer0 = network.Peer("Org2", "peer0")
+
+		_, cancel = context.WithTimeout(context.Background(), network.EventuallyTimeout)
+
+		marblechaincodeutil.AddMarble(network, orderer, channelID, chaincode.Name, `{"name":"test-marble-0", "color":"blue", "size":35, "owner":"tom", "price":99}`, org2Peer0)
+	})
+
+	AfterEach(func() {
+		cancel()
+
+		if process != nil {
+			process.Signal(syscall.SIGTERM)
+			Eventually(process.Wait(), network.EventuallyTimeout).Should(Receive())
+		}
+		if network != nil {
+			network.Cleanup()
+		}
+		os.RemoveAll(testDir)
+	})
+
+	// 6. The purge transaction takes effect only if the corresponding capability is set
+	//    - Add a few keys into a collection
+	//    - Issue a purge transaction without setting the new capability
+	//    - [Verify that the purge fails]
+	PIt("should fail with an error if the purge capability has not been enabled on the channel")
+
+	It("should prevent purged data being included in responses after the purge transaction has been committed", func() {
+		marblechaincodeutil.AssertPresentInCollectionM(network, channelID, chaincode.Name, `test-marble-0`, org2Peer0)
+		marblechaincodeutil.AssertPresentInCollectionMPD(network, channelID, chaincode.Name, "test-marble-0", org2Peer0)
+
+		marblechaincodeutil.PurgeMarble(network, orderer, channelID, chaincode.Name, `{"name":"test-marble-0"}`, org2Peer0)
+
+		marblechaincodeutil.AssertDoesNotExistInCollectionM(network, channelID, chaincode.Name, `test-marble-0`, org2Peer0)
```

Comment:
This assertion just checks the presence of marble in the private state. In addition, you would need to add the code for testing that the data is not returned in a query for retrieving BlockWithPrivatedata (which retrieves the private data from private data store.)

Reply:
I had a chat with @mbwhite and added a placeholder to make sure purged private data is not included in block event listening replays after the transaction has been committed. Hopefully this PR is in a reasonable state for merging now.